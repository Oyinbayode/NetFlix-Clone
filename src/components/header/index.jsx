import React from "react";
import { Link as ReactRouterLink } from "react-router-dom";

import {
  Background,
  Container,
  Logo,
  ButtonLink,
  Feature,
  FeatureCallOut,
  Text,
  TLink,
  Group,
  Picture,
  Profile,
  Dropdown,
} from "./styles/header";

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.TextLink = function HeaderTextLink({ children, ...restProps }) {
  return <TLink {...restProps}>{children}</TLink>;
};

Header.FeatureCallOut = function HeaderFeatureCallOut({
  children,
  ...restProps
}) {
  return <FeatureCallOut {...restProps}>{children}</FeatureCallOut>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

Header.ButtonLink = function HeaderButtonLink({ to, children, restProps }) {
  return (
    <ButtonLink to={to} {...restProps}>
      {children}
    </ButtonLink>
  );
};

Header.Group = function HeaderGroup({ children, ...restProps }) {
  return <Group {...restProps}>{children}</Group>;
};

Header.Picture = function HeaderPicture({ src, ...restProps }) {
  return <Picture {...restProps} src={`/images/users/${src}.png`} />;
};

Header.Profile = function HeaderProfile({ children, ...restProps }) {
  return <Profile {...restProps}>{children}</Profile>;
};

Header.Dropdown = function HeaderDropdown({ children, ...restProps }) {
  return <Dropdown {...restProps}>{children}</Dropdown>;
};
