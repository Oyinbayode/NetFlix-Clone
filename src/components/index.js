export { default as Jumbotron } from "./jumbotron";
export { default as Footer } from "./footer";
export { default as Accordion } from "./accordion";
export { default as OptForm } from "./opt-form";
export { default as Header } from "./header";
export { default as Feature } from "./feature";
export { default as SigninForm } from "./sign-in-form";
export { default as Profiles } from "./profiles";
export { default as Loading } from "./loading";
export { default as Card } from "./card";
export { default as Player } from "./player";
