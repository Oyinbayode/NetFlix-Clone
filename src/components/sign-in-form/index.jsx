import React from "react";
import {
  Container,
  Error,
  Base,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from "./styles/sign-in-form";

export default function SigninForm({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
}

SigninForm.Error = function SigninFormError({ children, ...restProps }) {
  return <Error {...restProps}>{children}</Error>;
};

SigninForm.Base = function SigninFormBase({ children, ...restProps }) {
  return <Base {...restProps}>{children}</Base>;
};

SigninForm.Title = function SigninFormTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

SigninForm.Text = function SigninFormText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

SigninForm.TextSmall = function SigninFormTextSmall({
  children,
  ...restProps
}) {
  return <TextSmall {...restProps}>{children}</TextSmall>;
};

SigninForm.Link = function SigninFormLink({ to, children, ...restProps }) {
  return (
    <Link to={to} {...restProps}>
      {children}
    </Link>
  );
};

SigninForm.Input = function SigninFormInput({ ...restProps }) {
  return <Input {...restProps} />;
};

SigninForm.Submit = function SigninFormSubmit({ children, ...restProps }) {
  return <Submit {...restProps}>{children}</Submit>;
};
