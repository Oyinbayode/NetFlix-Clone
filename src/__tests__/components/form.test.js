import React from "react";
import { render, screen } from "@testing-library/react";
import { SigninForm } from "../../components";

jest.mock("react-router-dom");

describe("<SignInForm />", () => {
  it("renders the <SignInForm /> with populated data", () => {
    const { container } = render(
      <SigninForm>
        <SigninForm.Title>Sign In Now</SigninForm.Title>
        <SigninForm.Base>
          <SigninForm.Input placeholder="Email Address" onChange={() => {}} />
          <SigninForm.Input
            placeholder="Password"
            type="password"
            onChange={() => {}}
          />
          <SigninForm.Submit disabled type="submit">
            Sign In
          </SigninForm.Submit>
        </SigninForm.Base>

        <SigninForm.Text>
          New to Netflix?{" "}
          {/* <SigninForm.Link to="/signup">Sign up now.</SigninForm.Link> */}
        </SigninForm.Text>

        <SigninForm.TextSmall>
          This page is protected by Google reCAPTCHA to ensure you are not a
          bot.
        </SigninForm.TextSmall>
      </SigninForm>
    );

    expect(
      screen.getByText(
        "This page is protected by Google reCAPTCHA to ensure you are not a bot."
      )
    ).toBeTruthy();
    expect(screen.getByText("Sign In")).toBeTruthy();
    expect(screen.getByText("Sign In Now")).toBeTruthy();
    expect(screen.getByText("Sign In").disabled).toBeTruthy();
    expect(screen.getByPlaceholderText("Email Address")).toBeTruthy();
    expect(screen.getByPlaceholderText("Password")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders the <Form /> with an error", () => {
    const { container } = render(
      <SigninForm>
        <SigninForm.Error>
          Your email address is already being used
        </SigninForm.Error>
        <SigninForm.Submit type="submit">Sign In</SigninForm.Submit>
      </SigninForm>
    );

    expect(
      screen.getByText("Your email address is already being used")
    ).toBeTruthy();
    expect(screen.queryByText("Sign In").disabled).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
