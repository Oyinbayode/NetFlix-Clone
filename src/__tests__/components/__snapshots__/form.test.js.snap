// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInForm /> renders the <Form /> with an error 1`] = `
<div
  class="sign-in-form__Container-sc-1clnpmu-0 kPUXXT"
>
  <div
    class="sign-in-form__Error-sc-1clnpmu-1 dDhGTI"
  >
    Your email address is already being used
  </div>
  <button
    class="sign-in-form__Submit-sc-1clnpmu-8 humHmQ"
    type="submit"
  >
    Sign In
  </button>
</div>
`;

exports[`<SignInForm /> renders the <SignInForm /> with populated data 1`] = `
<div
  class="sign-in-form__Container-sc-1clnpmu-0 kPUXXT"
>
  <h1
    class="sign-in-form__Title-sc-1clnpmu-3 cXEsUd"
  >
    Sign In Now
  </h1>
  <form
    class="sign-in-form__Base-sc-1clnpmu-2 bjZrYy"
  >
    <input
      class="sign-in-form__Input-sc-1clnpmu-7 egfjov"
      placeholder="Email Address"
    />
    <input
      class="sign-in-form__Input-sc-1clnpmu-7 egfjov"
      placeholder="Password"
      type="password"
    />
    <button
      class="sign-in-form__Submit-sc-1clnpmu-8 humHmQ"
      disabled=""
      type="submit"
    >
      Sign In
    </button>
  </form>
  <p
    class="sign-in-form__Text-sc-1clnpmu-4 iydtYY"
  >
    New to Netflix?
     
  </p>
  <p
    class="sign-in-form__TextSmall-sc-1clnpmu-5 iXpBBo"
  >
    This page is protected by Google reCAPTCHA to ensure you are not a bot.
  </p>
</div>
`;
